src/ports/postgres/modules/deep_learning/test/madlib_keras.sql_in

```diff
@@ -173,3 +172,36 @@ SELECT madlib_keras_predict(
     'x',
     $$ optimizer=SGD(lr=0.01, decay=1e-6, nesterov=True), loss='categorical_crossentropy', metrics=['accuracy']$$::text,
     'cifar10_predict');
+
+-- Validate that prediction output table exists and has correct schema
+SELECT assert(UPPER(atttypid::regtype::TEXT) = 'INTEGER', 'id column should be INTEGER type')
+    FROM pg_attribute WHERE attrelid = 'cifar10_predict'::regclass
+        AND attname = 'id';
+
+SELECT assert(UPPER(atttypid::regtype::TEXT) =
+    'DOUBLE PRECISION', 'prediction column should be DOUBLE PRECISION type')
+    FROM pg_attribute WHERE attrelid = 'cifar10_predict'::regclass
+        AND attname = 'prediction';
+
+-- Validate correct number of rows returned.
+SELECT assert(COUNT(*) = 2, "Output table of madlib_keras_predict should have two rows") FROM cifar10_predict;
+
+-- First test that all values are in set of class values; if this breaks, it's definitely a problem.
+SELECT assert(prediction in (0,1),'Predicted value not in set of defined class values for model') FROM cifar10_predict;
+
+-- Then test that each of the two images is correctly predicted.  If this breaks, it's likely a different problem, or
+   -- possibly due to the test model we chose not being trained well enough.  Based on current testing, it looks
+   -- reliable.
+SELECT assert(prediction=0,'Predicted value not in set of defined class values for model') FROM cifar10_predict WHERE id=1;
+SELECT assert(prediction=1,'Predicted value not in set of defined class values for model') FROM cifar10_predict WHERE id=2;
```

Comment:
Might not be a great idea to assert for exact prediction values. The model learnt in this test scenario is most likely something random, and we may end up seeing different results for different runs. We can instead assert that it is one of 0 or 1, like in the test in line 190.

Reply:
I actually tried running it multiple times, and it did have different predictions. The assert failed some times.

Reply:
This turned out to be a bug due to keras_out getting overwritten.  It was using the wrong output table.  As long as it uses the one from the first call to keras_fit (which runs for 3 iterations instead of just 1) the validation accuracy is always 100%.

I've fixed this now and added a test of keras_fit to ensure the accuracy is 100% after training.  Since there are only 2 training examples, if it can't classify them into 2 different classes by 3 iterations, I think that should be taken as a sign that something's broken.